include/counter.h

```diff
@@ -329,6 +330,7 @@ static inline int counter_cancel_channel_alarm(struct device *dev,
  * @retval -EBUSY if any alarm is active.
  */
 static inline int counter_set_top_value(struct device *dev, u32_t ticks,
+					bool reset,
```

Comment:
i wonder if it's not worth to bundle options to cfg structure like in `counter_set_channel_alarm`. It's then easier to add new features without braking backward compatibility (as long as false in option means legacy behavior). Now we exceed boundary of 4 arguments.

Reply:
I think that with 5 arguments we are still more efficient that using separate data structure.